docs/src/userguide/language_basics.rst

```diff
@@ -45,61 +48,104 @@ C variable and type definitions
 ===============================
 
 The :keyword:`cdef` statement is used to declare C variables, either local or
-module-level::
+module-level
+
+.. tabs::
+
+    .. group-tab:: Pure Python
+
+        .. code-block:: python
+
+            import cython
+
+            i: cython.int
+            j: cython.int
+            k: cython.int
+            f: cython.float
+            g: cython.int[42]
+            h: cython.p_float
+
+
+
+    .. group-tab:: Cython
+
+        .. code-block:: cython
 
-    cdef int i, j, k
-    cdef float f, g[42], *h
+            cdef int i, j, k
+            cdef float f, g[42], *h
 
 and C :keyword:`struct`, :keyword:`union` or :keyword:`enum` types:
 
-.. literalinclude:: ../../examples/userguide/language_basics/struct_union_enum.pyx
+.. tabs::
 
-See also :ref:`struct-union-enum-styles`
+    .. group-tab:: Pure Python
 
-.. note::
+        .. literalinclude:: ../../examples/userguide/language_basics/struct_union_enum.py
 
-    Structs can be declared as ``cdef packed struct``, which has
-    the same effect as the C directive ``#pragma pack(1)``.
+        .. NOTE:: Currently, Pure Python mode does not support enums.
 
-Declaring an enum as ``cpdef`` will create a :pep:`435`-style Python wrapper::
+    .. group-tab:: Cython
 
-    cpdef enum CheeseState:
-        hard = 1
-        soft = 2
-        runny = 3
+        .. literalinclude:: ../../examples/userguide/language_basics/struct_union_enum.pyx
 
+        See also :ref:`struct-union-enum-styles`
 
+        .. note::
 
-There is currently no special syntax for defining a constant, but you can use
-an anonymous :keyword:`enum` declaration for this purpose, for example,::
+            Structs can be declared as ``cdef packed struct``, which has
+            the same effect as the C directive ``#pragma pack(1)``.
```

Comment:
Looks like there's no way to do this in Python syntax. I created #4253 to track it.